app/src/main/java/com/space/quiz_app/presentation/base/fragment/QuizBaseFragment.kt

```diff
@@ -0,0 +1,59 @@
+package com.space.quiz_app.presentation.base.fragment
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.findNavController
+import com.space.quiz_app.presentation.base.view_model.QuizBaseViewModel
+import com.space.quiz_app.presentation.navigation.NavigationCommand
+import com.space.quiz_app.presentation.utils.observeNonNull
+import org.koin.androidx.viewmodel.ext.android.viewModelForClass
+import kotlin.reflect.KClass
+
+abstract class QuizBaseFragment<VM : QuizBaseViewModel>() : Fragment() {
+
+    abstract val viewModelClass: KClass<VM>
+    protected val viewModel: VM by viewModelForClass(clazz = viewModelClass)
+
+    protected abstract val layout: Int
+
+    abstract fun onBind()
+    abstract fun onCreateFragment()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(layout, container, false)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        onCreateFragment()
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        onBind()
+        observeNavigation()
+        findNavController().popBackStack()
+    }
```

Comment:
findNavController().popBackStack() why ?

Reply:
so when I press back button on home screen, it doesn't go back to login page but exits from app

Reply:
what if you press back button on quiz screen, doesn't have same behaviour ?

Reply:
No, it won't. During quiz it will initial exit dialog